### PR TITLE
RavenDB namespace changes in V4

### LINKED ISSUE #1474: missing namespace changes in raven upgrade guide
http://docs.particular.net/nservicebus/ravendb/upgrades/3to4#namespace-change

for example this commit https://github.com/Particular/docs.particular.net/commit/9555a7ae69d4b0648468c0944a4ae8f532ac9bf5

### PR DESCRIPTION
**Pull request by DavidBoike**
Fixes #1474 

Will remove WIP once final namespace move is complete and either 1) I'm sure that namespaces in samples don't need to be updated, or 2) I have appended a commit to update the samples to match.

### CODE REVIEW

**Comment by DavidBoike** [comment]
Only namespace change unaccounted for was to an obsoleted class, which samples/snippets should not reference. This PR is ready for review.

File: nservicebus/upgrades/ravendb-3to4.md [hunk 1]
```diff
@@ -14,9 +14,20 @@ redirects:
 As part of this update [NServiceBus Version 6](/nservicebus/upgrades/5to6.md) will be required.
 
 
-## Namespace change
+## Namespace changes
 
-The `RavenDBPersistence` has moved from `NServiceBus.Persistence` namespace to `NServiceBus`. There is no need to have an extra using statement if the NServiceBus.RavenDB NuGet package is installed. 
+Namespaces for public types have been consolidated to make customizations more discoverable. The primary namespaces are `NServiceBus` for customization options that need to be discoverable, and `NServiceBus.Persistence.RavenDB` for items off the beaten path. If the NServiceBus.RavenDB NuGet package is installed, a single `using NServiceBus` directive should be sufficient to find all necessary options.
```
Comment (SimonCropp) [inline nservicebus/upgrades/ravendb-3to4.md]: "Off the beaten path " => advanced apis

Comment (SimonCropp) [inline nservicebus/upgrades/ravendb-3to4.md]: If the package is installed... Isnt that redundant in an upgrade guide

File: nservicebus/upgrades/ravendb-3to4.md [hunk 2]
```diff
@@ -14,9 +14,20 @@ redirects:
...
+As part of this move, the following classes were moved to different namespaces:
+
+* `NServiceBus.Persistence.RavenDBPersistence` to the `NServiceBus` namespace
+* `NServiceBus.RavenDB.Outbox.RavenDBOutboxExtensions` to the `NServiceBus` namespace
+* `NServiceBus.RavenDB.ConnectionParameters` to the `NServiceBus.Persistence.RavenDB` namespace
+
+
+## Use of RavenDB Async API
+
+NServiceBus now uses the asynchronous RavenDB API for all operations. If sharing the session between NServiceBus and handler code is required, then handler code will need to be adjusted to utilize the asynchronous RavenDB API as well.
```
Comment (SimonCropp) [inline nservicebus/upgrades/ravendb-3to4.md]: Code snippet? Link to old session type and new session type?

Comment (DavidBoike) [inline nservicebus/upgrades/ravendb-3to4.md]: You mean links to Raven doco for session and async session, correct?

Comment (SimonCropp) [inline nservicebus/upgrades/ravendb-3to4.md]: Correct

**Comment by DavidBoike** [comment]
Updated.

The Raven docs are awful for linking directly to a class or interface, so I linked directly to stable GitHub source links which document the operations of IDocumentSession/IAsyncDocumentSession quite well via XML doc comments.

**Pull request merged**